sys/linux/auto.txt

```diff
@@ -185,320 +187,301 @@ type msghdr_tcp_metrics_auto[CMD, POLICY] msghdr_netlink[netlink_msg_t[genl_tcp_
+mq_timedsend_time64$auto(mqdes int32, u_msg_ptr ptr[inout, int8], msg_len int32, msg_prio int32, u_abs_timeout ptr[inout, __kernel_timespec$auto_record]) (automatic)
+mq_unlink$auto(u_name ptr[inout, int8]) (automatic)
+mremap$auto(addr int32, old_len int32, new_len int32, flags int32, new_addr int32) (automatic)
+mseal$auto(start int32, len int32, flags int32) (automatic)
+msgctl$auto(msqid int32, cmd int32, buf ptr[inout, msqid_ds$auto_record]) (automatic)
+msgget$auto(key int32, msgflg int32) (automatic)
+msgrcv$auto(msqid int32, msgp ptr[inout, msgbuf$auto_record], msgsz int32, msgtyp int32, msgflg int32) (automatic)
+msgsnd$auto(msqid int32, msgp ptr[inout, msgbuf$auto_record], msgsz int32, msgflg int32) (automatic)
+msync$auto(start int32, len int32, flags int32) (automatic)
+munlock$auto(start int32, len int32) (automatic)
+munmap$auto(addr int32, len int32) (automatic)
+name_to_handle_at$auto(dfd int32, name ptr[inout, int8], handle ptr[inout, file_handle$auto_record], mnt_id ptr[inout, int32], flag int32) (automatic)
+nanosleep$auto(rqtp ptr[inout, __kernel_timespec$auto_record], rmtp ptr[inout, __kernel_timespec$auto_record]) (automatic)
+newfstatat$auto(dfd int32, filename ptr[inout, int8], statbuf ptr[inout, stat$auto_record], flag int32) (automatic)
+nice$auto(increment int32) (automatic)
+oldfstat$auto(fd int32, statbuf ptr[inout, __old_kernel_stat$auto_record]) (automatic)
+oldlstat$auto(filename ptr[inout, int8], statbuf ptr[inout, __old_kernel_stat$auto_record]) (automatic)
+oldolduname$auto(name ptr[inout, oldold_utsname$auto_record]) (automatic)
+oldstat$auto(filename ptr[inout, int8], statbuf ptr[inout, __old_kernel_stat$auto_record]) (automatic)
+olduname$auto(name ptr[inout, old_utsname$auto_record]) (automatic)
+open$auto(filename ptr[inout, int8], flags int32, mode int16) (automatic)
+open_by_handle_at$auto(mountdirfd int32, handle ptr[inout, file_handle$auto_record], flags int32) (automatic)
+open_tree$auto(dfd int32, filename ptr[inout, int8], flags int32) (automatic)
+openat$auto(dfd int32, filename ptr[inout, int8], flags int32, mode int16) (automatic)
+openat2$auto(dfd int32, filename ptr[inout, int8], how ptr[inout, open_how$auto_record], usize int32) (automatic)
+perf_event_open$auto(attr_uptr ptr[inout, perf_event_attr$auto_record], pid int32, cpu int32, group_fd int32, flags int32) (automatic)
+personality$auto(personality int32) (automatic)
+pidfd_getfd$auto(pidfd int32, fd int32, flags int32) (automatic)
+pidfd_open$auto(pid int32, flags int32) (automatic)
+pidfd_send_signal$auto(pidfd int32, sig int32, info ptr[inout, siginfo$auto_record], flags int32) (automatic)
+pipe$auto(fildes ptr[inout, int32]) (automatic)
+pipe2$auto(fildes ptr[inout, int32], flags int32) (automatic)
+pivot_root$auto(new_root ptr[inout, int8], put_old ptr[inout, int8]) (automatic)
```

Comment:
Please add 2 TODOs in the code:
1. Use `string` for `char*`/`const char*` args.
2. Use `in` direction for const pointers (e.g. `const char*` should be `ptr[in, string]`).

Reply:
3. Infer if non-const pointer is in or inout. (should be possible by looking if it's used in copy_to/from_user).

Reply:
1 and 2 are done. I added a TODO for 3.